### PR TITLE
feat(sdk): SDK theming part 5 - success, summarize, warning, white, text-white, bg-white

### PR DESCRIPTION
**Pull request by deniskaber**
Relates to https://github.com/metabase/metabase/issues/42694
Also is a small part of https://www.notion.so/metabase/100-Align-approach-to-Mantine-styles-customization-1066f77c43694c36a6e90a3d2374b731

### Description

Migrate hardcoded colors used in emotion styles:
- success
- summarize
- warning
- white

### How to verify

1. CI should pass
2. Visual tests should return same results

### CODE REVIEW

File: .storybook/preview-head.html [hunk 1]
```diff
@@ -7,5 +7,9 @@
     --mb-color-focus: #cbe2f7;
     --mb-color-bg-dark: #93a1ab;
     --mb-color-bg-light: #f9fbfc;
+    --mb-color-bg-white: #fff;
+    --mb-color-text-white: #fff;
+    --mb-color-success: #84bb4c;
+    --mb-color-summarize: #88bf4d;
```
Comment (WiNloSt) [inline .storybook/preview-head.html:13]: why doesn't `warning` get the same css custom property as other colors? I see it's only used in a handful of places, but it's still a semantic color.

File: .storybook/preview-head.html [hunk 2]
```diff
@@ -7,5 +7,9 @@
     --mb-color-focus: #cbe2f7;
     --mb-color-bg-dark: #93a1ab;
     --mb-color-bg-light: #f9fbfc;
+    --mb-color-bg-white: #fff;
+    --mb-color-text-white: #fff;
```
Comment (WiNloSt) [inline .storybook/preview-head.html:11]: Is there a need to differentiate these colors? Wouldn't having a single `--mb-color-white` be easier to understand as both refer to pure white?

For comparison, I believe we used to have `text-brand` and now we just use `brand` directly.

File: enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts [hunk 3]
```diff
@@ -39,8 +39,8 @@ export const DEFAULT_METABASE_COMPONENT_THEME: MetabaseComponentTheme = {
   },
   pivotTable: {
     rowToggle: {
-      textColor: "white",
-      backgroundColor: "text-light",
+      textColor: "text-white",
+      backgroundColor: "text-light", // TODO: should it be "bg-dark" ?
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:43]: Agree, let's not use color that's meant for another purpose here.

File: enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts [hunk 4]
```diff
@@ -39,8 +39,8 @@ export const DEFAULT_METABASE_COMPONENT_THEME: MetabaseComponentTheme = {
   },
   pivotTable: {
     rowToggle: {
-      textColor: "white",
-      backgroundColor: "text-light",
+      textColor: "text-white",
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:42]: I made a comment in `.storybook/preview-head.html` I think `textColor: "white"` and `backgroundColor: "white"` is easier to understand imo. Having 2 of the same colors are just confusing.

We're different from tailwind in the sense that tailwind uses `bg` prefix to denote where that color would go. In our case, the `bg-<color>` variant [doesn't ever have the same color as the normal `<color>` counterpart](https://github.com/metabase/metabase/blob/54d9fb225705f5493e9c9860c9a2b7a837399ecf/frontend/src/metabase/lib/colors/palette.ts#L37-L44) (expect the `white` color).

if you search the use of `color("white")` 99% is for `background-color` and `color`. So I don't get it why we need to have 3 white colors

- `white`
- `text-white`
- `bg-white`

when `white` is also mostly used for both `color` and `background-color` anyway.

Comment (WiNloSt) [reply enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:42]: [To close the loop](https://metaboat.slack.com/archives/C057WD5L0JG/p1717676739175609?thread_ts=1717663992.584869&cid=C057WD5L0JG). Let's just keep `white` and remove `bg-white` and `text-white`. I don't think it's gonna be too hard to use find and replace `bg-white` to `white` and `text-white` to `white`. But please let me know what you think.

Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:42]: I think the only time we should be differentiating between `bg-` and `text-` is if we're dealing with variants like `light`, `medium`, and `dark`. If it's a base color we should have one thing. If we change the base color (i.e. we want a dark theme and we set `white` to a different color, they should all have the same transformation

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.styled.tsx [hunk 5]
```diff
@@ -24,8 +24,8 @@ export const FieldSettingsButtonsContainer = styled.div`
 
 export const WarningBanner = styled.div`
   padding: ${space(2)};
-  border: 1px solid ${color("warning")};
+  border: 1px solid ${() => color("warning")};
```
Comment (oisincoveney) [inline frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.styled.tsx:27]: do we need to add a function here?

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx [hunk 6]
```diff
@@ -53,7 +53,7 @@ export const InteractiveEmbeddingCTA = () => {
       >
         <Text
           align="center"
-          c="white"
+          c="text-swhite"
```
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:56]:
```suggestion
          c="text-white"
```
Happens to the best of us 🤪

File: frontend/src/metabase/setup/components/NewsletterForm/NewsletterForm.styled.tsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ export const EmailFormLabelCard = styled.div`
   display: flex;
   padding: 0 1.5rem;
   color: ${color("text-medium")};
-  background-color: ${color("white")};
+  background-color: var(--mb-color-bg-white);
```
Comment (oisincoveney) [inline frontend/src/metabase/setup/components/NewsletterForm/NewsletterForm.styled.tsx:29]: I've noticed a few times we convert from a `color()` call to a `var` css class, and the other way around as well. We also have the Mantine theme calls for colors as well. What should we try to default to in the future?

**Comment by oisincoveney** [comment]
It looks good, just a few questions. I personally feel ok with approving this but since it's a broad PR it's probably best to finish off that slack conversation and come to an obvious conclusion before merging this

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorSuggestions/ExpressionEditorSuggestions.styled.tsx [hunk 8]
```diff
@@ -27,8 +27,6 @@ export const ExpressionListItem = styled.li<{ isHighlighted: boolean }>`
   padding-right: 0.5rem;
   cursor: pointer;
   height: 2rem;
-  display: flex;
-  align-items: center;
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorSuggestions/ExpressionEditorSuggestions.styled.tsx:31]: Awesome removing these duplicate attributes.

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorSuggestions/ExpressionEditorSuggestions.styled.tsx [hunk 9]
```diff
@@ -50,7 +48,6 @@ export const SuggestionTitle = styled.span`
 `;
 
 export const PopoverHoverTarget = styled(BasePopoverHoverTarget)`
-  padding: 0;
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorSuggestions/ExpressionEditorSuggestions.styled.tsx:53]: 👍

File: frontend/src/metabase/query_builder/components/view/sidebars/ChartTypeSidebar.styled.tsx [hunk 10]
```diff
@@ -9,7 +9,7 @@ export interface OptionRootProps {
 
 const getOptionIconColor = ({ isSelected }: OptionIconContainerProps) => {
   if (isSelected) {
-    return color("white");
+    return color("text-white");
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/sidebars/ChartTypeSidebar.styled.tsx:12]: Use CSS custom property so it's consistent with the change on line 59

```suggestion
    return "var(--mb-color-text-white)";
```

File: frontend/src/metabase/setup/components/InactiveStep/InactiveStep.styled.tsx [hunk 11]
```diff
@@ -14,12 +14,11 @@ export const StepRoot = styled.section<Props>`
   padding: 1rem 2rem;
   margin-bottom: 1.75rem;
   background-color: ${props =>
-    props.isCompleted ? color("white") : "var(--mb-color-bg-medium)"};
+    color(props.isCompleted ? "bg-white" : "bg-medium")};
```
Comment (WiNloSt) [inline frontend/src/metabase/setup/components/InactiveStep/InactiveStep.styled.tsx:17]: I'd leave `--mb-color-bg-medium` as is as it should be preferable over any color functions calls

File: frontend/src/metabase/setup/components/InactiveStep/InactiveStep.styled.tsx [hunk 12]
```diff
@@ -14,12 +14,11 @@ export const StepRoot = styled.section<Props>`
...
 `;
 
 export const StepTitle = styled.div<Props>`
-  color: ${props =>
-    props.isCompleted ? color("success") : "var(--mb-color-brand)"};
+  color: ${props => (props.isCompleted ? color("success") : color("brand"))};
```
Comment (WiNloSt) [inline frontend/src/metabase/setup/components/InactiveStep/InactiveStep.styled.tsx:21]: I think we should keep using `--mb-color-brand`

File: frontend/src/metabase/setup/components/InactiveStep/InactiveStep.styled.tsx [hunk 13]
```diff
@@ -36,11 +35,10 @@ export const StepLabel = styled.div<Props>`
   width: 2.625rem;
   height: 2.625rem;
   border: 1px solid
-    ${props =>
-      props.isCompleted ? color("success") : "var(--mb-color-border)"};
+    ${props => (props.isCompleted ? color("success") : color("border"))};
```
Comment (WiNloSt) [inline frontend/src/metabase/setup/components/InactiveStep/InactiveStep.styled.tsx:38]: I think we should keep using `--mb-color-border`

File: frontend/src/metabase/setup/components/LanguageStep/LanguageStep.styled.tsx [hunk 14]
```diff
@@ -34,14 +34,14 @@ export interface LocaleContainerProps {
 export const LocaleButton = styled.span<LocaleContainerProps>`
   display: block;
   padding: 0.5rem;
-  color: ${props => color(props.checked ? "white" : "text-dark")};
+  color: ${props => color(props.checked ? "text-white" : "text-dark")};
   border-radius: 0.25rem;
   background-color: ${props =>
-    props.checked ? "var(--mb-color-brand)" : color("white")};
+    props.checked ? color("brand") : color("bg-white")};
```
Comment (WiNloSt) [inline frontend/src/metabase/setup/components/LanguageStep/LanguageStep.styled.tsx:40]: Should keep using `--mb-color-brand`?

File: frontend/src/metabase/ui/components/navigation/NavLink/NavLink.styled.tsx [hunk 15]
```diff
@@ -26,14 +26,14 @@ export const getNavLinkOverrides = (): MantineThemeOverride["components"] => ({
             },
 
             backgroundColor: theme.fn.themeColor("brand"),
-            color: "white",
+            color: theme.fn.themeColor("text-white"),
```
Comment (WiNloSt) [inline frontend/src/metabase/ui/components/navigation/NavLink/NavLink.styled.tsx:29]: since this is a mantine override settings. I assume the color name would work right out of the gate e.g.
```suggestion
            color: "text-white",
```

Can you double check if this is working as intended?

File: frontend/src/metabase/visualizations/components/LeafletChoropleth.jsx [hunk 16]
```diff
@@ -49,7 +49,7 @@ const LeafletChoropleth = ({
         fillColor: getColor(feature),
         weight: 1,
         opacity: 1,
-        color: "white",
+        color: getColor("white"),
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/components/LeafletChoropleth.jsx:52]: I don't think this is how `getColor` is meant to be used. You can see that it accepts `feature` not a color.

https://github.com/metabase/metabase/blob/0768495c06cb1e8a35ab704d26b6f3f58ac48f13/frontend/src/metabase/visualizations/components/ChoroplethMap.jsx#L354-L357


I think we should revert this change.

File: enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts (hunk 4 above)
Comment (WiNloSt) [reply enterprise/frontend/src/embedding-sdk/lib/theme/default-component-theme.ts:42]: Yeah that's right @oisincoveney Here's the [decision from another thread.](https://metaboat.slack.com/archives/C063Q3F1HPF/p1717684958978589?thread_ts=1717677247.911649&cid=C063Q3F1HPF)

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.styled.tsx (hunk 5 above)
Comment (WiNloSt) [reply frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.styled.tsx:27]: That's explained here. https://github.com/metabase/metabase/pull/43512#discussion_r1627510527

File: frontend/src/metabase/ui/components/navigation/NavLink/NavLink.styled.tsx (hunk 15 above)
Comment (oisincoveney) [reply frontend/src/metabase/ui/components/navigation/NavLink/NavLink.styled.tsx:29]: I think for consistency I like the idea of using the `theme.fn.themeColor` for both of them. But yeah let's check if that still works

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.styled.tsx (hunk 5 above)
Comment (oisincoveney) [reply frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormCreator.styled.tsx:27]: awesome, thanks :smile:

File: enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx [hunk 17]
```diff
@@ -59,7 +59,7 @@ export const StyledLauncher = styled(Flex)<
   font-weight: ${({ forRoot, inheritsRootStrategy }) =>
     forRoot || inheritsRootStrategy ? "normal" : "bold"};
   background-color: ${({ forRoot }) =>
-    forRoot ? "var(--mb-color-bg-medium)" : color("white")};
+    forRoot ? color("bg-medium") : color("bg-white")};
```
[the remark this answers is not in the document]
Comment (WiNloSt) [reply enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx:62]: That's understandable. At this point. Let's just keep it as is, so we could merge this PR sooner rather than later.

File: frontend/src/metabase/admin/datamodel/components/GuiQueryEditor/GuiQueryEditor.module.css [hunk 18]
```diff
@@ -29,5 +29,5 @@
 /* create a smaller inset arrow on the right */
 .arrowRight::after {
   right: -19px;
-  border-left-color: var(--color-white);
+  border-left-color: var(--mb-color-bg-white);
```
[the remark this answers is not in the document]
Comment (WiNloSt) [reply frontend/src/metabase/admin/datamodel/components/GuiQueryEditor/GuiQueryEditor.module.css:32]: Just to close the loop. I think it makes sense that the border follows the background color since they both are connected. And it's been used throughout the PR.

**Review by WiNloSt — APPROVED** [review]
